File: Programmers/Level1&2/숫자_블록.py
def solution(begin, end):
    def find(n):
        if n == 1:
            return 0
        for i in range(2, int(n ** 0.5) + 1):
            div, mod = divmod(n, i)
            if div > 10000000:
                continue
            if mod == 0:
                return div
        return 1

    answer = []
    for n in range(begin, end + 1):
        answer.append(find(n))
    return answer


print(solution(1, 30))

File: 프로그래머스/2/12923. 숫자 블록/숫자 블록.java
import java.util.*;

class Solution {

    private int find(long n) {
        if (n == 1) {
            return 0;
        }

        int result = 1;

        for (long x = 2; x * x <= n; x++) {
            if (n % x == 0) {
                result = (int) x;
                if (n / x <= 10000000) {
                    return (int) (n / x);
                }
            }
        }

        return result;
    }

    public int[] solution(long begin, long end) {
        int size = (int) (end - begin + 1);
        int[] answer = new int[size];

        for (long n = begin; n <= end; n++) {
            answer[(int) (n - begin)] = find(n);
        }

        return answer;
    }
}